### ISSUE TITLE
Merge Envoy mobile code into the Envoy repo

### ISSUE THREAD

**Issue by alyssawilk** [envoyproxy/envoy#23758]
Almost all files are copied from envoy-mobile repo to envoy/mobile  Source of truth for mobile changes is still envoy-mobile.  @alyssawilk is doing updates until we flip source of truth to envoy/mobile.

TODO: 
- run relevant Envoy workflows when dependencies are bumped (phlax)

Thursday(ish) switch source of truth from envoy-mobile to envoy/mobile
 - copy over PRs which have landed since then
 - notify on various slack channels / notify on PRs in flight

Post-move
- fix up / move github release cron jobs
-  Circleci functionality (e.g. docs) https://github.com/envoyproxy/envoy/pull/24295
- Update references to envoyproxy/envoy-mobile repo in source code and docs (Tackle after Thursday)
- Archive envoyproxy/envoy-mobile repo (Tackle after release workflows moved over)

**Comment by alyssawilk** [comment]
Known missing things
-- swift lint because I can't get it to work.
-- perf (https://github.com/envoyproxy/envoy/pull/24153)
-- release validation 
-- release

-- all cron workflows (weekly release, release, etc)

**Comment by alyssawilk** [comment]
perf turned up in  (https://github.com/envoyproxy/envoy/pull/24153)

**Comment by alyssawilk** [comment]
@jpsim @Augustyniak I think we're at a point we should flip source of truth to the Envoy repo, and do a weekly copy into the E-M repo until we fix up tagging and releases.  I've snagged some time on JP's calendar to talk about it next week on the assumption he'll find a time that works and invite whoever is relevant on the Lyft side.

**Comment by jpsim** [comment]
Some other migration tasks that come to mind:

* Update references to `envoyproxy/envoy-mobile` repo in source code and docs
* Enable `/retest` for GitHub Actions: https://github.com/envoyproxy/envoy/issues/24156
* Archive `envoyproxy/envoy-mobile` repo

**Comment by Augustyniak** [comment]
Related fix: https://github.com/envoyproxy/envoy/pull/24228

**Comment by jpsim** [comment]
> run relevant Envoy workflows when dependencies are bumped (phlax)

Done in https://github.com/envoyproxy/envoy/pull/24407

**Comment by jpsim** [comment]
> fix up / move github release cron jobs

This still needs to be done. We discussed this at last week's Envoy Mobile community call, but didn't come to a clear conclusion.

> Archive envoyproxy/envoy-mobile repo (Tackle after release workflows moved over)

This was mostly done, GH Actions were disabled, a note was added to the readme saying the repo is archived.

I haven't gone and fully marked the repo as archive using GitHub's tools because that would also lock down issues, and we're still using those for the time being.
